linera-base/src/crypto/ed25519.rs

```diff
@@ -312,10 +311,9 @@ impl Ed25519Signature {
     where
         T: BcsSignable<'de>,
     {
-        let mut message = Vec::new();
-        value.write(&mut message);
+        let preimage = CryptoHash::new(value).as_bytes().0;
```

Comment:
Did you mean `prehash`?

Do we need to change the `new` implementation for the two other algorithms as well to avoid unnecessary double hashing?

Reply:
Yes, `prehash`. Other impls should already match - `secp256k1` was doing that already and for EIP-191